Fix attribute regex matching

thebluepotato opened the pull request:
It used to work well but since using Swift 3 (or so it seems), the attributeRegex only matches css attributes without values. For example, "a[rel]" is correctly interpreted as ".//a[@rel]" but "a[rel='next']" isn't matched so it returns only ".//a", therefore catching all anchors in the HTML document. Because you can put quite anything as a value of an attribute, matching any character inside the brackets seems to be the right solution.
I tested it with various attribute names and values and it seems to transform them into XPath quite fine.

cezheng commented:
@thebluepotato 
Hi, thanks for the PR. The CSS to XPath conversion logic, which is taken from Ono, is not yet comprehension at the moment, and I definitely want to make it able to convert more kinds of selectors.

Currently the tests for CSS to XPath conversion are [here](https://github.com/cezheng/Fuzi/blob/master/FuziTests/CSSTests.swift). Pleasure share the test cases you have tried.

And for the PR itself, using `.` here might not seem to be a good idea to me, since the `]` will also be included. And since Regex matching is by default greedy, which means if you have multiple attribute selectors, this regex will match the first open bracket with the last close bracket.

thebluepotato commented:
You're welcome! I've changed the Regex to match everything but `[]` and I've added two tests for attributes with values and multiple attributes in CSS selectors. All tests pass!

cezheng commented:
@thebluepotato Thanks. I think this should be good for now. More comprehensive conversion will be added when I have some free time.

cezheng commented:
It seems xctool is having problems running tests with Xcode 8, I will merge it anyway

thebluepotato commented:
I'm currently studying for my finals but in a week or two I'll have a lot of free time and I'll gladly look at the CSS -> XPath conversions to make them more capable! Cheers

cezheng commented:
Thanks man, I appreciate it. Good luck on your finals. :smile:
